code/datums/components/bump_attack.dm

```diff
@@ -0,0 +1,57 @@
+/datum/component/bump_attack
+    var/active = FALSE
+    var/datum/action/bump_attack_toggle/toggle_action
+
+/datum/component/bump_attack/Initialize()
+	. = ..()
+	if(!ismovableatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(isliving(parent))
+		toggle_action = new()
+		RegisterSignal(toggle_action, COMSIG_ACTION_TRIGGER, .proc/bump_attack_toggle)
+		toggle_action.give_action(parent)
+		toggle_action.update_button_icon(active)
+	else
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/bump_attack/proc/bump_attack_toggle(datum/source, atom/target)
+	var/mob/living/bumper = parent
+
+	active = !active
+	to_chat(bumper, "<span class='notice'>You will now [active ? "attack" : "push"] enemies who are in your way.</span>")
```

Comment:
Description is somewhat misleading. pushing would probably depend on intent.
``to_chat(bumper, "<span class='notice'>You will [active ? "now" : "no longer"] attack enemies who are in your way.</span>")`` probably makes more sense.